src/main/java/io/stargate/sgv2/jsonapi/service/processor/MeteredCommandProcessor.java

```diff
@@ -0,0 +1,116 @@
+  private final Tag defaultErrorClass;
+
+  @Inject
+  public MeteredCommandProcessor(
+      CommandProcessor commandProcessor,
+      MeterRegistry meterRegistry,
+      StargateRequestInfo stargateRequestInfo,
+      JsonApiMetricsConfig jsonApiMetricsConfig,
+      MetricsConfig metricsConfig) {
+    this.commandProcessor = commandProcessor;
+    this.meterRegistry = meterRegistry;
+    tenantConfig = metricsConfig.tenantRequestCounter();
+    customMetricsConfig = jsonApiMetricsConfig.customMetricsConfig();
+    this.stargateRequestInfo = stargateRequestInfo;
+    errorTrue = Tag.of(tenantConfig.errorTag(), "true");
+    errorFalse = Tag.of(tenantConfig.errorTag(), "false");
+    tenantUnknown = Tag.of(tenantConfig.tenantTag(), UNKNOWN_VALUE);
+    defaultErrorCode = Tag.of(customMetricsConfig.errorCode(), NA);
+    defaultErrorClass = Tag.of(customMetricsConfig.errorClass(), NA);
+  }
+
+  /**
+   * Processes a single command in a given command context.
+   *
+   * @param commandContext {@link CommandContext}
+   * @param command {@link Command}
+   * @return Uni emitting the result of the command execution.
+   * @param <T> Type of the command.
+   */
+  public <T extends Command> Uni<CommandResult> processCommand(
+      CommandContext commandContext, T command) {
+    Timer.Sample sample = Timer.start(meterRegistry);
+    // start by resolving the command, get resolver
+    return commandProcessor
+        .processCommand(commandContext, command)
+        .onItem()
+        .invoke(
+            result -> {
+              Tags tags = getCustomTags(command, result);
+              // add metrics
+              sample.stop(meterRegistry.timer(customMetricsConfig.metricsName(), tags));
+            });
+  }
+
+  /**
+   * Generate custom tags based on the command and result.
+   *
+   * @param command - request command
+   * @param result - response command result
+   * @return
+   */
+  private Tags getCustomTags(Command command, CommandResult result) {
+    Tag commandTag = Tag.of(customMetricsConfig.command(), command.getClass().getSimpleName());
+    String tenant = stargateRequestInfo.getTenantId().orElse(UNKNOWN_VALUE);
+    Tag tenantTag = Tag.of(tenantConfig.tenantTag(), tenant);
+    Tag errorTag = errorFalse;
+    Tag errorClassTag = defaultErrorClass;
+    Tag errorCodeTag = defaultErrorCode;
+    // if error is present, add error tags else use defaults
+    if (null != result.errors() && !result.errors().isEmpty()) {
+      errorTag = errorTrue;
+      String errorClass =
+          (String) result.errors().get(0).fields().getOrDefault("exceptionClass", UNKNOWN_VALUE);
+      errorClassTag = Tag.of(customMetricsConfig.errorClass(), errorClass);
+      String errorCode =
+          (String) result.errors().get(0).fields().getOrDefault("errorCode", UNKNOWN_VALUE);
+      errorCodeTag = Tag.of(customMetricsConfig.errorCode(), errorCode);
+    }
+    Tags tags = Tags.of(commandTag, tenantTag, errorTag, errorClassTag, errorCodeTag);
```

Comment:
I would only have `errorTag` set to true if it's only real server side error, this way everything is set to errors, so we can not use alerting here.. Is it possible to do it here?

Reply:
Will create a separate PR, we are not currently tracking gRPC error in the ErrorCode enum.